src/ZoneServer/Scripting/AI/AiScript.cs

```diff
@@ -304,11 +304,16 @@ protected ICombatEntity GetMostHated()
 				var handle = entry.Key;
 				var hate = entry.Value;
 
-				if (hate > highestHate)
-				{
-					highestHate = hate;
-					mostHated = this.Entity.Map.GetCombatEntity(handle);
-				}
+				if (hate < highestHate)
```

Comment:
You're right, when I looked at it I didn't think it would matter, but it is true that it results in checking the buffs more often, so I've changed it to <=.